rfc/0014-metadata.md

```diff
@@ -0,0 +1,616 @@
+
+## Examples / User Stories
+[examples]: #examples
+
+## Publishing a policy on Artifact Hub for the first time
+
+As a policy author, I want to publish my policy on Artifact Hub. My policy has
+already a `metadata.yml` file, however I have to copy-paste, adapt and expand
+the contents of `metadata.yml` into the `artifacthub-pkg.yml` file.
+
+I would rather have a way to automatically generate the `artifacthub-pkg.yml` file.
+
+## Tag a new release of a policy
+
+As a policy author, I'm about to publish a new version of my policy. This policy
+is also registered on Artifact Hub.
+
+I would like to have some automation that updates the contents of the already
+existing `artifacthub-pkg.yml` file.
+
+# Detailed design
+[design]: #detailed-design
+
+Each Kubewarden policy must have a `metadata.yml` file. While only certain ones
+are going to have a `artifacthub-pkg.yml` file.
+
+We want to have the following workflow:
+
+* User creates a `metadata.yml` file. This file is scaffolded when the policy
+  is created for the 1st time
+* User enters proper data inside of `metadata.yml`
+* User keeps annotating the policy in the usual way: `kwctl annotate --metadata-path `metadata.yml`
+
+If the user wants to publish the policy on Artifact Hub, we will offer the following
+workflow:
+
+* User extends the `metadata.yml` file with some annotations that are specific
+  to Artifact Hub
+* User invokes: `kwctl scaffold artifacthub --metadata-path metadata.yml`.
+  This command will generate the `artifacthub-pkg.yml` file
+
+The freshly created `kwctl scaffold artifacthub` can be run as many times
+as the user wants. The contents of `artifacthub-pkg.yml` are going to be
+overwritten.
+
+It's important to point out that a user can completely avoid this workflow and
+keep managing the `artifacthub-pkg.yml` file as he prefers.
+
+## Comparing `metadata.yml` and `artifacthub-pkg.yml`
+
+The following snippet is taken from one of our policies, it
+includes all the mandatory fields:
+
+```yaml
+rules:
+- apiGroups: [""]
+  apiVersions: ["v1"]
+  resources: ["pods"]
+  operations: ["CREATE", "UPDATE"]
+mutating: true
+contextAware: false
+executionMode: kubewarden-wapc
+annotations: {}
+```
+
+The `annotations` key holds a dictionary where both keys and values
+are strings. It's a free form dictionary, that allows users to store
+arbitrary data.
+
+There are however some recommended fields:
+
+```yaml
+annotations:
+  io.kubewarden.policy.title: verify-image-signatures
+  io.kubewarden.policy.description: A Kubewarden Policy that verifies all the signatures of the container images referenced by a Pod
+  io.kubewarden.policy.author: "Raul Cabello Martin <raul.cabello@suse.com>, Victor Cuadrado Juan <vcuadradojuan@suse.de>"
+  io.kubewarden.policy.url: https://github.com/kubewarden/verify-image-signatures
+  io.kubewarden.policy.source: https://github.com/kubewarden/verify-image-signatures
+  io.kubewarden.policy.license: Apache-2.0
```

Comment:
as pointed out in https://github.com/kubewarden/echo/issues/11#issuecomment-1397346695,
we could add a `rancher/hide-ui` annotation, analogous to [the one used for rancher charts](https://github.com/kubewarden/helm-charts/blob/main/charts/kubewarden-crds/Chart.yaml#L33).

Reply:
Yes, but I would not mention that explicitly inside of the RFC.

All the annotations that being with `io.artifacthub.<something>: <value>` would be translated to

```yaml
annotations:
  <something>: <value>
```

inside of the `artifacthub-pkg.yml` file.

Reply:
fair enough, resolving!